src/backend/pagemap.h

```diff
@@ -236,12 +236,12 @@ namespace snmalloc
      * read/write.
      */
     template<bool potentially_out_of_range>
-    const T& get(address_t p)
+    T& get_mut(address_t p)
```

Comment:
In the past, I think there was objection to such an accessor because it couldn't be used across the sandbox boundary.

Now, I'm a little jittery about returning mutable references to the default value.  That seems risky?

Reply:
So to use the pagemap storage for the buddy allocator, then we need to a way to modify it.  This is currently done in `set_next` in the address space manager, but it gets more complex for the RBTree as it does rotations and a lot more mutations.  Using the pagemap across a boundary like that is not a good idea.

There will be two versions of the buddy allocator, one using in band meta-data and one using the pagemap.  In the compartment version, we could just us in band meta-data.

It is a tricky balance between making the best allocator for X and the best allocator for Y.